### ISSUE TITLE
🐛  Link Experience card focus issue

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#8107]
## 🐛 Bug

When using the “link an experience” dialog, after saving, focus is not returned to the trigger properly, it goes back up to the “Skip to main content” link. This happens when creating as well as editing linked experiences.

## 🦋 Expected Behaviour

Focus should be returned to the button that triggered the modal in the first place.

## 🕵️ Details

Add any additional details that could assist with troubleshooting/fixing the issue.

- **Operating System**: Windows and MacOS
- **Browser**: Chrome, Firefox
- **Adaptive Technology**: NVDA, VoiceOver, Keyboard 

## 📋 Steps to Reproduce

1. Log into UAT using an account that already has skills associated. (If you don't have an account with skills, use this testing account: Username: WandaM1964 PW: T4l3nt$!
2. Go to Skill library using a screen reader  (NVDA, VoiceOver). 
3. Activate the skill name to edit it. 
4. Tab down to the experience section (or use the landmark shortcut). 
5. Open the experience dialogue. 
6. Dismiss it immediately.

## 📸 Screenshot

(https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbEx4IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--1a227b488079b5dbd13826d73517edf98cbedbf3/image.png)

(https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbFR4IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--859f086bd12d431d09dde17aaed95d2e439e4147/image.png)

## 🙋‍♀️ Proposed Solution

We have one dialogue that is associated with multiple buttons. So, should we associate each button with a single dialogue, or evaluate whether we can associate multiple buttons with a single dialogue? Can we apply this [Radix Primitive solution](https://www.radix-ui.com/primitives/docs/components/dialog#custom-apis). 

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] Criteria 1
- [ ] Criteria 2

**Comment by petertgiles** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/GC-Digital-Talent-6092dc05157fa30010edaef9/issues/GCTC-NTGC/gc-digital-talent/8107?planning-poker) with Zenhub @brindasasi @esizer @JamesHuf @mnigh @tristan-orourke @vd1992 @yonikid15
